src/devices/cpu/gigatron/gigatron.cpp

```diff
@@ -95,36 +97,173 @@ void gigatron_cpu_device::init()
 	m_x = 0;
 	m_y = 0;
 	m_pc = 0;
+	m_npc = (m_pc + 1);
+	m_inReg = 0xFF;
 	state_add(GTRON_AC,        "AC",        m_ac);
 	state_add(GTRON_X,         "X",         m_x);
 	state_add(GTRON_Y,         "Y",         m_y);
 
 	set_icountptr(m_icount);
 }
 
-void gigatron_cpu_device::branchOp(int op, int mode, int bus, int d)
+void gigatron_cpu_device::branchOp(uint8_t op, uint8_t mode, uint8_t bus, uint8_t d)
 {
+	const uint8_t ZERO = 0x80;
+	uint8_t c = 0;
+	uint8_t ac2 = m_ac ^ ZERO;
+	uint16_t base = m_pc & 0xff00;
+	switch (mode) {
+		case 0:
+			c = true;
+			base = m_y << 8;
+			break;
+		case 1:
+			c = (ac2 > ZERO);
+			break;
+		case 2:
+			c = (ac2 < ZERO);
+			break;
+		case 3:
+			c = (ac2 != ZERO);
+			break;
+		case 4:
+			c = (ac2 == ZERO);
+			break;
+		case 5:
+			c = (ac2 >= ZERO);
+			break;
+		case 6:
+			c = (ac2 <= ZERO);
+			break;
+		case 7:
+			c = true;
+			break;
+	}
+	
+	if (c) {
+		uint8_t b = offset(bus, d);
+		m_npc = base | b;
+	}
 }
 
-void gigatron_cpu_device::aluOp(int op, int mode, int bus, int d)
+void gigatron_cpu_device::aluOp(uint8_t op, uint8_t mode, uint8_t bus, uint8_t d)
 {
-	int b;
-	(void)b;
+	uint8_t b = 0;
+	//(void)b;
 	switch(bus) {
 		case 0:
 			b = d;
 			break;
 		case 1:
+			b = gigatron_readmem8((uint16_t)(addr(mode, d) & m_ramMask));
+			break;
 		case 2:
 			b = m_ac;
 			break;
 		case 3:
+			b = m_inReg;
+			break;
+	}
+	switch(op) {
+		case 1:
+			b = (m_ac & b);
+			break;
+		case 2:
+			b = (m_ac | b);
+			break;
+		case 3:
+			b = (m_ac ^ b);
+			break;
+		case 4:
+			b = (m_ac + b);
+			break;
+		case 5:
+			b = (m_ac - b);
+			break;
+	}
+	switch (mode) {
+		case 0:
+		case 1:
+		case 2:
+		case 3:
+			m_ac = b;
 			break;
+		case 4:
+			m_x = b;
+			break;
+		case 5:
+			m_y = b;
+			break;
+	}
+}
+
+uint16_t gigatron_cpu_device::addr(uint8_t mode, uint8_t d)
+{
+	switch (mode) {
+		case 0:
+		case 4:
+		case 5:
+		case 6:
+			return d;
+		case 1:
+			return m_x;
+		case 2:
+			return (m_y << 8) | d;
+		case 3:
+			return (m_y << 8) | m_x;
+		case 7:
+			uint16_t addr2 = (m_y << 8) | m_x;
+			m_x = (m_x + 1) & 0xff;
+			return addr2;
 	}
+	return 0;
 }
 
-void gigatron_cpu_device::storeOp(int op, int mode, int bus, int d)
+uint8_t gigatron_cpu_device::offset(uint8_t bus, uint8_t d)
 {
+	switch (bus) {
+		case 0:
+			return d;
+		case 1:
+			return gigatron_readmem8(d);
+		case 2:
+			return m_ac;
+		case 3:	
+			return m_inReg;
+	}
+	return 0;
+}
+
+void gigatron_cpu_device::storeOp(uint8_t op, uint8_t mode, uint8_t bus, uint8_t d)
+{
+	uint8_t b = 0;
+	switch (bus) {
+		case 0:
+			b = d;
+			break;
+		case 1:
+			b = 0;
+			logerror("UNDEFINED BEHAVIOR! %b", m_pc);
```

Comment:
What is "%b"? This will give a compiler error. Like I said before, please test/compile your work!

Reply:
I confused it with %d, and now I realize that it will not do what I was looking for it to do at all.

Changing it to 0x%04x\n